File: mcdowell/python/ch5-02.py


def decToBinString(n):

    if n <= 0 or n >= 1.:
        return "ERROR"

    text = "."
    while n > 0:
        if len(text) > 32:
            return "ERROR"
        double = n * 2
        if double >= 1.:
            text += '1'
            n = double - 1
        else:
            text += '0'
            n = double

    return text



print decToBinString(.25)

File: mcdowell/php/ch5-02.php
<?php


function decToBinString($n) {
    if ($n >= 1 || $n <= 0) {
        return "ERROR";
    }

    $text = ".";
    while ($n > 0) {
        if (strlen($text) > 32) {
            //return "ERROR";
        }
        $r = $n * 2;
        if ($r >= 1) {
            $text .= '1';
            $n = $r - 1;
        }
        else {
            $text .= '0';
            $n = $r;
        }
    }
    return $text;
}


echo decToBinString(.75);

